x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/Pivot.java

```diff
@@ -77,6 +79,11 @@ public void validateConfig(ActionListener<Boolean> listener) {
         listener.onResponse(true);
     }
 
+    @Override
+    public List<String> getPerformanceCriticalFields() {
+        return config.getGroupConfig().getGroups().values().stream().map(SingleGroupSource::getField).collect(toList());
```

Comment:
I am not sure all of these would be performance critical. it almost seems like they would be performance critical IF they didn't support missing buckets right?

If the group config supported missing buckets, we don't get the benefit of filtering for changes or something? I remember there was a trade off. 

Right @hendrikmuhs

Reply:
> If the group config supported missing buckets, we don't get the benefit of filtering for changes or something? I remember there was a trade off.

missing bucket runs a bit slower in composite aggs, but I think we can ignore that, its like 10-20% improvement. Problematic are group_by with scripts, but we already have a warning for that: We warn if change detection is completely disabled, which means we do a full re-run on every checkpoint.